src/schema/sources.ts

```diff
@@ -325,7 +342,10 @@ export const ensureSourcePermissions = async (
     const source = await ctx.con
       .getRepository(Source)
       .findOneByOrFail([{ id: sourceId }, { handle: sourceId }]);
-    if (await canAccessSource(ctx, source, permission)) {
+    const sourceMember = await ctx.con
```

Comment:
I see we already fetch the relevant source member inside `canAccessSource` so maybe instead of fetching it here we can move the condition from the switch case to inside the if. We also have to make sure the source is a squad just like we do to post